### DISCUSSION TITLE
Liquidsoap and DSP plugins

### DISCUSSION THREAD

**Discussion by gavin2812** [savonet/liquidsoap#3070] [General]
Hi all,

Would it be possible to use Liquidsoap with any of the legacy Winamp type DSP/Effects plugins? I'm thinking maybe some of the limiter/compressor/processor type plugins?

Not sure if this is already possible or could be considered? While I know many of them are quite old and not updated, there are quite a few that are very good still.

If already possible, is there any documentation for this about how to implement? Thinking specifically about applying some of the limiter/compressor type plugins to a live stream.

If not possible, could it be?

Thanks.

**Comment by toots** [comment]
I think it could be fun but this would be limited by time and available documentation. In terms of plugins I believe our first most important one to support would be VST plugins. However, we would welcome any contribution adding support for winamp DSP plugins.

Are you aware of any specs for them? I had a quick look around and couldn't find any..

**Comment by gavin2812** [comment]
Hi @toots,

Not really, no. I found the following which may (or not) be relevant:

http://wiki.winamp.com/wiki/Plug-in_Developer 
http://wiki.winamp.com/wiki/SDK_Contents (which contains a link to what looks like a Windows SDK if some sort (http://download.nullsoft.com/winamp/plugin-dev/WA5.55_SDK.exe)

I guess in an ideal world one would simply put one of these plugins in a 'plugin' folder and have liquidsoap load it as Winamp would have done in the past.

It would certainly be cool to be able to reuse some of these with liquidsoap but my technical knowledge is not good enough to know where to start with something like this.

Just a thought!
